VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/EqualsAndGetHashcodeNotImplementedTogether/EqualsAndGetHashcodeNotImplementedTogetherAnalyzer.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using VSDiagnostics.Utilities;
+
+namespace VSDiagnostics.Diagnostics.General.EqualsAndGetHashcodeNotImplementedTogether
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class EqualsAndGetHashcodeNotImplementedTogetherAnalyzer : DiagnosticAnalyzer
+    {
+        private const DiagnosticSeverity Severity = DiagnosticSeverity.Warning;
+
+        private static readonly string Category = VSDiagnosticsResources.GeneralCategory;
+        private static readonly string Message = VSDiagnosticsResources.EqualsAndGetHashcodeNotImplementedTogetherAnalyzerMessage;
+        private static readonly string Title = VSDiagnosticsResources.EqualsAndGetHashcodeNotImplementedTogetherAnalyzerTitle;
+
+        internal static DiagnosticDescriptor Rule
+            => new DiagnosticDescriptor(DiagnosticId.EqualsAndGetHashcodeNotImplementedTogether, Title, Message, Category, Severity, true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.ClassDeclaration);
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var classDeclaration = (ClassDeclarationSyntax)context.Node;
+
+            var equalsImplemented = false;
+            var getHashcodeImplemented = false;
+
+            foreach (var node in classDeclaration.Members)
+            {
+                if (!node.IsKind(SyntaxKind.MethodDeclaration))
+                {
+                    continue;
+                }
+
+                var methodDeclaration = (MethodDeclarationSyntax)node;
+
+                if (!IsOverride(methodDeclaration))
+                {
+                    continue;
+                }
+
+                if (methodDeclaration.Identifier.ValueText == nameof(Equals) && methodDeclaration.ParameterList.Parameters.Count == 1)
+                {
+                    equalsImplemented = true;
+                }
+
+                if (methodDeclaration.Identifier.ValueText == nameof(GetHashCode) && methodDeclaration.ParameterList.Parameters.Count == 0)
+                {
+                    getHashcodeImplemented = true;
+                }
+            }
+
+            if (equalsImplemented ^ getHashcodeImplemented)
+            {
+                context.ReportDiagnostic(Diagnostic.Create(Rule, classDeclaration.GetLocation(),
+                    ImmutableDictionary.CreateRange(new[] { new KeyValuePair<string, string>("IsEqualsImplemented", equalsImplemented.ToString()) })));
+            }
+        }
+
+        private bool IsOverride(MethodDeclarationSyntax methodDeclaration)
```

Comment:
This isn't a `Contains` call,  the Linq version would be `return methodDeclaration.Modifiers.Any(modifier => modifier.IsKind(SyntaxKind.OverrideKeyword));`